lib/voicebase/v3/client.rb

```diff
@@ -0,0 +1,153 @@
+module VoiceBase
+  module V3
+    module Client
+      def self.extended(client, args = {})
+        client.api_host     = client.args[:host] || ENV.fetch('VOICEBASE_V3_API_HOST', 'https://apis.voicebase.com')
+        client.api_endpoint = client.args[:api_endpoint] || ENV.fetch('VOICEBASE_V3_API_ENDPOINT', '/v3')
+      end
+
+      def authenticate!
+        auth = {:username => @auth_key, :password => @auth_secret}
+        response = VoiceBase::Response.new(
+          self.class.get(
+            uri + '/access/users/admin/tokens',
+            basic_auth: auth,
+            headers: {
+              'User-Agent'   => @user_agent,
+              'Accept'       => 'application/json'
+            }
+          ), api_version)
+        @token = VoiceBase::Client::Token.new(response.tokens.any? && response.tokens.first.fetch("token"))
+      rescue NoMethodError => ex
+        raise VoiceBase::AuthenticationError, response.status_message
```

Comment:
are we sure we have a `response`? should we then put the `response` outside the `rescue` block. 

Maybe we don't even need a `rescue`, if between the `response` and `@token` lines, we `raise VoiceBase::AuthenticationError if something?(response)`

Reply:
I actually didn't implement this. it's from v2.  https://github.com/usertesting/voicebase-client-ruby/blob/master/lib/voicebase/v2/client.rb#L13-L25
Maybe this should be removed and let V3 should extend V2.